_posts/en/newsletters/2019-09-25-newsletter.md

```diff
@@ -0,0 +1,136 @@
+---
+title: 'Bitcoin Optech Newsletter #65'
+permalink: /en/newsletters/2019/09/25/
+name: 2019-09-25-newsletter
+slug: 2019-09-25-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter requests feedback on not allowing P2SH-wrapped
+addresses for taproot, describes proposed changes to script resource
+limits in tapscript, and mentions a discussion about watchtower storage
+costs.  Also included are a selection of popular questions and answers
+from the Bitcoin StackExchange and a short list of notable changes to
+popular Bitcoin infrastructure projects.
+
+{% comment %}<!-- include references.md below the fold but above any Jekyll/Liquid variables-->{% endcomment %}
+{% include references.md %}
+
+## Action items
+
+- **Comment if you expect to need P2SH-wrapped taproot addresses:**
+  recent [discussion][p2sh taproot] on the Bitcoin-Dev mailing list
+  indicates that the [bip-taproot][] proposal may be amended to disallow
+  creating taproot inputs by paying P2SH outputs (the way you can
+  currently use P2WPKH and P2WSH inputs created from P2SH outputs).
+  Anyone who expects to need P2SH-wrapped taproot should describe their
+  planned usecase on the mailing list or by contacting the bip-taproot
+  author privately.
+
+## News
+
+- **Tapscript resource limits:** the [bip-tapscript][] proposal limits
+  transactions to one signature-checking operation (sigop) for every 12.5 vbytes
+  the witness data adds to the size of the transaction (plus one free
+  sigop per input).  Because signatures are expected to be 16.0 vbytes,
+  this limit prevents abuse without affecting normal users.  Compared to
+  earlier abuse-prevention solutions, it means any valid taproot
+  transaction can be included in a block regardless of how many sigops
+  it contains---keeping miner transaction code simple and fast.
+
+    In a mailing list [post][tapscript limits], bip-tapscript author
+    Pieter Wuille notes that he and Andrew Poelstra examined other
+    resource limits on scripts that were put in place to prevent nodes
+    from using an excessive amount of CPU or memory during verification.
+    He describes some of their findings and advocates the following rule
+    changes:
+
+    > * Replace the separate sigops counter with a "executed sigops must
+    >   not exceed (witness size / [12.5 vbytes]) + 1" rule (already in the BIP).
+    > * Drop the 10,000 byte limit for script size (and 3,600 byte
+    >   standardness limit)
+    > * Drop the 201 non-push ops limit per script.
+    > * Drop the 100 input stack elements standardness limit, and
+    >   replace with a (consensus) 1,000 limit.
+
```

Comment:
What is the unit of the 1,000 limit?

Reply:
That sentence does state the unit "stack elements".